mpmp/utilities/file_utilities.py

```diff
@@ -2,27 +2,83 @@
 Functions for writing and processing output files
 """
 from pathlib import Path
+import pickle as pkl
 
 import pandas as pd
 
 from mpmp.exceptions import ResultsFileExistsError
 
-def make_output_dir(results_dir, identifier, exp_string='cancer_type'):
+def make_output_dir(experiment_dir, identifier):
     """Create a directory to write output to."""
-    output_dir = Path(results_dir, exp_string, identifier).resolve()
+    output_dir = Path(experiment_dir, identifier).resolve()
     output_dir.mkdir(parents=True, exist_ok=True)
     return output_dir
 
 
+def construct_filename(output_dir,
+                       identifier,
+                       shuffle_labels,
+                       model_options,
+                       file_descriptor,
+                       extension):
+    """Construct a filename from experimental parameters.
+
+    This allows us to standardize what information goes into the filename
+    (i.e. what information constitutes a unique "experiment").
+    """
+    training_data = model_options.training_data
+    signal = 'shuffled' if shuffle_labels else 'signal'
+    seed = model_options.seed
+    if identifier is None:
+        # this might be the case for files that pertain to the whole experiment
+        # (e.g. metadata/parameters file)
+        return Path(output_dir,
+                    '{}_s{}_{}{}'.format(training_data,
+                                         seed,
+                                         file_descriptor,
+                                         extension))
```

Comment:
If using Python version >= 3.6, you can use f-strings here if you like them:

```suggestion
        return Path(output_dir,
                    f'{training_data}_s{seed}_{file_descriptor}{extension}')
```

Reply:
I should update all of my scripts to use these - created #12 for this.